Sources/Utility/ArgumentParser.swift

```diff
@@ -382,12 +378,104 @@ public final class ArgumentParser {
         }
     }
 
+    /// An enum representing way to parse argument values.
+    public enum ArrayParsingStrategy {
+        /// Will parse only the next argument and append all values together: `-Xcc -Lfoo -Xcc -Lbar`
+        case oneByOne
+        /// Will parse all values up to the next option argument: `--files file1 file2 --verbosity 1`
+        case upToNextOption
+        /// Will parse all remaining arguments, usually for executable commands: `swift run exe --option 1`
+        case remaining
+
+        /// Function that parses the current arguments iterator based on the strategy and
+        /// returns the parsed values.
+        func parse(with parser: inout Parser, kind: ArgumentKind.Type) throws -> [ArgumentKind] {
+            var values: [ArgumentKind] = []
+
+            switch self {
+            case .oneByOne:
+                var parserProtocol = parser as ArgumentParserProtocol
+                try values.append(kind.init(parser: &parserProtocol))
+                parser = parserProtocol as! Parser
+            case .upToNextOption:
+                /// Iterate over arguments until the end or an optional argument
+                while true {
+                    let previousIterator = parser.argumentsIterator
+
+                    /// If there are no more arguments, we can stop parsing
+                    guard let nextArgument = try? parser.next() else {
+                        break
+                    }
+
+                    /// Parse the next argument only if it is non-optional
+                    if isPositional(argument: nextArgument) {
+                        try values.append(kind.parse(argument: nextArgument))
+                    } else {
+                        /// If we have reached an optional argument, we need to back-track by
+                        /// resetting the iterator to before the last `next` so that the
+                        /// top-level parser can continue from this argument.
+                        parser.argumentsIterator = previousIterator
+                        break
+                    }
+                }
+            case .remaining:
+                while let nextArgument = try? parser.next() {
```

Comment:
Hm, can we add a variant that doesn't throw? It feels wrong to "iterate until an error is throw".
Make we can rename the throwing version to `pop()` and make `next()` return optional.